### PR TITLE
API assignment first

### PR DESCRIPTION
**Pull request by tejashrighogare**

### CODE REVIEW

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package APIAssignment;
+
+import io.restassured.*;
```
Comment (vrushalee123) [inline src/test/java/APIAssignment/APIAssignmentGet.java:4]: Please make it maven project. I can't see pom.xml

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+public class APIAssignmentGet {
+    @Test
+    public void getAllProductList() {
+        var productListURL = "https://automationexercise.com/api/productsList";
+        Response responseVariable = get(productListURL);
+        // validate status code
+        var statusCodeValue = responseVariable.getStatusCode();
+        Assert.assertEquals(200, statusCodeValue);
+        System.out.println("status code is "+statusCodeValue);
```
Comment (vrushalee123) [inline src/test/java/APIAssignment/APIAssignmentGet.java]: Remove sysouts

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        //validate content of list
+        RestAssured.baseURI = productListURL;
+        RequestSpecification specification = RestAssured.given();
+        Response response = specification.request(Method.GET);
+        ResponseBody body = response.body();
+        var bodyContent = body.asString();
+        Assert.assertEquals(bodyContent.contains("name"), true);
+        System.out.println("name present in my response");
+        //Validate List returned by API
+        var getALLProductList = given().when().get(productListURL).then().log().all().toString();
```
Comment (vrushalee123) [inline src/test/java/APIAssignment/APIAssignmentGet.java:37]: Validate content

**Review by vrushalee123 — CHANGES_REQUESTED** [review]

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        var productListURL = "https://automationexercise.com/api/productsList";
+        Response responseVariable = get(productListURL);
+        // validate status code
+        var statusCodeValue = responseVariable.getStatusCode();
+        Assert.assertEquals(200, statusCodeValue);
+        System.out.println("status code is "+statusCodeValue);
+        //validate content of list
+        RestAssured.baseURI = productListURL;
+        RequestSpecification specification = RestAssured.given();
+        Response response = specification.request(Method.GET);
```
Comment (pooja-hasure) [inline src/test/java/APIAssignment/APIAssignmentGet.java]: You can add Query or path parameter to get the particular response  on line no.23

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        ResponseBody body = response.body();
+        var bodyContent = body.asString();
+        Assert.assertEquals(bodyContent.contains("name"), true);
```
Comment (pooja-hasure) [inline src/test/java/APIAssignment/APIAssignmentGet.java]: You can first verify the status code and status line.

File: src/test/java/APIAssignment/APIAssignmentGet.java (hunk 2 above)
Comment (tejashrighogare) [reply src/test/java/APIAssignment/APIAssignmentGet.java]: I will remove

File: src/test/java/APIAssignment/APIAssignmentGet.java (hunk 3 above)
Comment (tejashrighogare) [reply src/test/java/APIAssignment/APIAssignmentGet.java:37]: Okay

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 6]
```diff
@@ -1,36 +1,68 @@
...
         ResponseBody body = response.body();
         var bodyContent = body.asString();
-        Assert.assertEquals(bodyContent.contains("name"), true);
-        System.out.println("name present in my response");
-        //Validate List returned by API
-        var getALLProductList = given().when().get(productListURL).then().log().all().toString();
-        System.out.println(getALLProductList);
-        // Validate Length
-        JsonPath jsonResponse = new JsonPath(response.asString());
+        var productName = response.jsonPath().get("products[3].name").toString();
```
Comment (vrushalee123) [inline src/test/java/APIAssignment/APIAssignmentGet.java]: Don't use hard coded index value.

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 7]
```diff
@@ -1,36 +1,68 @@
...
+        var productPrice = response.jsonPath().get("products[3].price").toString();
+        var productBrand = response.jsonPath().get("products[3].brand").toString();
+        Assert.assertEquals(productName, "Stylish Dress");
+        Assert.assertEquals(productPrice, "Rs. 1500");
```
Comment (vrushalee123) [inline src/test/java/APIAssignment/APIAssignmentGet.java]: Write code to validate all contents.

**Review by vrushalee123 — CHANGES_REQUESTED** [review]

File: src/test/java/dataProvider/ConfigReader.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+public class ConfigReader {
+    public static Properties readPropertiesFile(String fileName) //throws IOException
+    {
+        FileInputStream fileStream = null;
+        Properties propertiesObject = null;
+        try {
+            fileStream = new FileInputStream(fileName);
+            propertiesObject = new Properties();
+            propertiesObject.load(fileStream);
+        } catch (FileNotFoundException exception) {
```
Comment (amitsahu8) [inline src/test/java/dataProvider/ConfigReader.java]: use or in Catch

File: src/test/java/dataProvider/ConfigReader.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+            exception.printStackTrace();
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
+        }
+        return propertiesObject;
+    }
+
+    public static String getURL() {
+        String configFilePath = "src\\test\\ConfigProperties\\configuration.properties";
```
Comment (amitsahu8) [inline src/test/java/dataProvider/ConfigReader.java]: don't hard code the path, use System.getProperty("user.dir")

File: src/test/java/dataProvider/ConfigReader.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
+package dataProvider;
+
+import java.io.*;
+import java.util.Properties;
+
+public class ConfigReader {
+    public static Properties readPropertiesFile(String fileName) //throws IOException
+    {
+        FileInputStream fileStream = null;
```
Comment (amitsahu8) [inline src/test/java/dataProvider/ConfigReader.java]: why FileInputStream declared here?

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        JsonPath jsonResponse = new JsonPath(response.asString());
+        logger.info("my json response: " + jsonResponse.get("products"));
+        var productsArray = jsonResponse.get("products");
+        var idLength = jsonResponse.getInt("products.id.size()");
+        var productName = ConfigReader.getURL("productName");
+        var price = ConfigReader.getURL("price");
+        var brand = ConfigReader.getURL("brand");
+        var category = ConfigReader.getURL("category");
+        var currentProductName = "";
+        for (var counter = 0; counter < productsArray.toString().length(); counter++) {
```
Comment (amitsahu8) [inline src/test/java/APIAssignment/APIAssignmentGet.java]: use lambda function

File: src/test/java/APIAssignment/APIAssignmentGet.java [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.log4j.*;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+public class APIAssignmentGet {
+    private String productListURL = "";
+    private Logger logger = Logger.getLogger(APIAssignmentGet.class.getName());
```
Comment (amitsahu8) [inline src/test/java/APIAssignment/APIAssignmentGet.java]: why .getName()?

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: src/test/java/APIAssignment/Product.java [hunk 13]
```diff
@@ -0,0 +1,15 @@
+package APIAssignment;
+
+public class Product {
+    public String productId;
+    public String productName;
+    public String productPrice;
+    public String productBrand;
+
+    public Product(String productListId, String productListName, String productListPrice, String productListBrand) {
+        productId = productListId;
```
Comment (amitsahu8) [inline src/test/java/APIAssignment/Product.java]: use getter setter to set and get the values

File: src/main/java/dataProvider/ConfigReader.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+    public static String getPropertyValue(String propertyName) {
+        String configFilePath = System.getProperty("user.dir");
+        Properties configProperties = ConfigReader.readPropertiesFile(configFilePath + "/configuration.properties");
+        return configProperties.getProperty(propertyName);
+    }
+
+    public static Properties readPropertiesFile(String fileName) {
+
+        try {
```
Comment (amitsahu8) [inline src/main/java/dataProvider/ConfigReader.java:20]: remove extra spaces and use or operator to combine the catch

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: src/test/java/APIAssignment/Product.java [hunk 15]
```diff
@@ -1,15 +1,44 @@
...
     public String productPrice;
     public String productBrand;
 
-    public Product(String productListId, String productListName, String productListPrice, String productListBrand) {
-        productId = productListId;
-        productName = productListName;
-        productPrice = productListPrice;
-        productBrand = productListBrand;
+    public void getProductId() {
+        System.out.println("Id=" + productId);
```
Comment (amitsahu8) [inline src/test/java/APIAssignment/Product.java]: Use logger instead of sysout and in getter return the variable

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

**Review by amitsahu8 — APPROVED** [review]

File: src/main/java/dataProvider/ConfigReader.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+        Properties configProperties = ConfigReader.readPropertiesFile(configFilePath + "/configuration.properties");
+        return configProperties.getProperty(propertyName);
+    }
+
+    public static Properties readPropertiesFile(String fileName) {
+        try {
+            FileInputStream fileStream = new FileInputStream(fileName);
+            propertiesObject = new Properties();
+            propertiesObject.load(fileStream);
+        } catch (Exception e) {
```
Comment (vrushalee123) [inline src/main/java/dataProvider/ConfigReader.java]: Change exception to specific expression and use logger to print errors properly.

**Review by vrushalee123 — CHANGES_REQUESTED** [review]

File: src/main/java/dataProvider/ConfigReader.java [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public static Properties readPropertiesFile(String fileName) {
+        try {
+
+            FileInputStream fileStream = new FileInputStream(fileName);
+            propertiesObject = new Properties();
+            propertiesObject.load(fileStream);
+        } catch (Exception FileNotFoundException) {
+
```
Comment (vrushalee123) [inline src/main/java/dataProvider/ConfigReader.java]: Remove extra space. Please do proper indentation.

**Review by vrushalee123 — CHANGES_REQUESTED** [review]
Please do proper formatting.

File: src/main/java/dataProvider/ConfigReader.java [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+        Properties configProperties = ConfigReader.readPropertiesFile(configFilePath + "/configuration.properties");
+        return configProperties.getProperty(propertyName);
+    }
+
+    public static Properties readPropertiesFile(String fileName) {
+        try {
+            FileInputStream fileStream = new FileInputStream(fileName);
+            propertiesObject = new Properties();
+            propertiesObject.load(fileStream);
+        } catch (Exception FileNotFoundException) {
```
Comment (vrushalee123) [inline src/main/java/dataProvider/ConfigReader.java]: Why two exceptions

Comment (vrushalee123) [reply src/main/java/dataProvider/ConfigReader.java]: Please make it FileNotFoundException

**Review by vrushalee123 — CHANGES_REQUESTED** [review]

**Review by vrushalee123 — APPROVED** [review]

**Pull request merged**